### ISSUE TITLE
@liveblocks/react-tiptap: components not displaying when using React compiler and/or shouldRenderOnTransaction: false

### ISSUE THREAD

**Issue by Ecaratti** [liveblocks/liveblocks#2055]
**Describe the bug**

When using either or both React compiler or shouldRenderonTransaction: true on Tiptap's useEditor, the FloatingComposer fails to appear.
 
When re-running my dev server without reactCompiler and with shouldRerenderOnTransaction: true, the component works as expected.

**To Reproduce**

Just set  shouldRerenderOnTransaction: false when using Tiptap's useEditor,
```typescript
  const editor = useEditor({
    extensions: [liveblocks, StarterKit],
    immediatelyRender: false,
    shouldRerenderOnTransaction: false,
  });
```

Issue is similar when setting 

```typescript
  experimental: {
    reactCompiler: true,
  },
``` 

in next.config.ts
/

**Expected behavior**

Floating composer should be displayed when triggering editor.chain().focus().addPendingComment().run()

**Environment (please complete the following information):**

- @tiptap 2.9.1.
- next 15.0.2
- @liveblocks 2.11.0

**Additional context**

The issue spawns from using:
```typescript
  const storage = editor?.storage.liveblocksComments as CommentsExtensionStorage | undefined;
  const showComposer = !!storage?.pendingCommentSelection;
``` 

in the [FloatingComposer](https://github.com/liveblocks/liveblocks/blob/main/packages/liveblocks-react-tiptap/src/comments/FloatingComposer.tsx) (lines 32-34).

Tiptap 2.5 introduced performance optimisations which requires the use of useEditorState() to get the latest editor state when shouldRenderTransaction is set to false (cf [announcement ](https://tiptap.dev/blog/release-notes/say-hello-to-tiptap-2-5-our-most-performant-editor-yet) and [doc](https://tiptap.dev/docs/examples/advanced/react-performance#page-title)))

Suggested fix:  in FloatingComposer replace showComposer by:

```typescript
const currentEditorState = useEditorState({
    editor,
    selector: (ctx) => ({
      showComposer: !!ctx.editor?.storage.liveblocksComments?.pendingCommentSelection,
    }),
    equalityFn: (prev, next) => {
      if (!next) return false;
      return prev.showComposer === next.showComposer;
    },
  });
const showComposer = currentEditorState?.showComposer;
``` 

This fixed the issue for me and seems to work either with or without the react compiler and with shouldRerenderOnTransaction either true or false.

It may be something you would need to consider for the other components of the @liveblocks/react-tiptap package. 

It may also be worthwhile mentionning in the documentation that when using TipTap's editor with shouldRerenderOnTransaction: false, things like editor.isActive("liveblocksCommentMark") won't work, and the user must use useEditorState(), as described above.

Cheers

**Comment by Ecaratti** [comment]
Same issue with [FloatingThreads](https://github.com/liveblocks/liveblocks/blob/main/packages/liveblocks-react-tiptap/src/comments/FloatingThreads.tsx)

Managed to get it working using useEditorState to get the pluginState (line 64). There may be some way to improve performance and unncessary re-renders by modifying the equalityFn, but I'll leave it at that for now. 

### before

```typescript
  const pluginState = editor ? THREADS_PLUGIN_KEY.getState(editor.state) as ThreadPluginState : null;
``` 

### after

```typescript
  const currentEditorState = useEditorState({
    editor,
    selector: (ctx) => {
      if (!ctx?.editor?.state) return null;
      //only necessary to get the plugin from the editor state because I don't have access to the plugin directly in my node_modules.
      //in the package use: const state = THREADS_PLUGIN_KEY.getState(ctx.editor.state) as ThreadPluginState
      const plugin = ctx.editor.state.plugins.find(
        //@ts-expect-error, this works but the type is not correct. Not sure why.
        (p) => p.key === "lb-threads-plugin$"
      );
      const state = plugin?.getState(ctx.editor.state);

      return {
        pluginState: state,
      };
    },
    equalityFn: (prev, next) => {
      if (!prev || !next) return false;
      return isEqual(prev.pluginState, next.pluginState);
    },
  });

  const pluginState = currentEditorState?.pluginState;
```

**Comment by jrowny** [comment]
Thanks so much for the suggestion, we will likely implement the useEditorState. I had come across this bug while testing with Blocknote and tried solving it with useSyncExternalStore, but this is much better than that.

**Comment by Ecaratti** [comment]
Fantastic, looking forward to it!
Also applied the same fix from FloatingThreads to AnchoredThreads and works like a charm.

**Comment by jrowny** [comment]
The PR is up, the fix should go out in 2.11.1. Thanks again for the report.
